Add issues to project action

dtzar opened the pull request:
This will make it so any time an issue is created in the repository that it will get automatically added with a `needs-triage` label and be added to the Notary-Planning project using the [Add To GitHub Projects Beta action](https://github.com/marketplace/actions/add-to-github-projects-beta).

dtzar commented:
@SteveLasker to make this work we need to add `ADD_TO_PROJECT_PAT` as a secret in this repository?  I don't have permissions to do so.
I have a PAT we could use or you can [generate your own](https://github.com/marketplace/actions/add-to-github-projects-beta#creating-a-pat-and-adding-it-to-your-repository)...

FeynmanZhou commented:
LGTM. Now we are using the [Notary planning Project Board](https://github.com/orgs/notaryproject/projects/10/views/2) to manage all of those issues.

SteveLasker reviewed (approved):
LGTM

shizhMSFT commented on .github/workflows/add-to-project.yml:
```diff
@@ -0,0 +1,16 @@
+      - uses: actions/add-to-project@main
+        with:
+          project-url: https://github.com/orgs/notaryproject/projects/10
+          github-token: ${{ secrets.ADD_TO_PROJECT_PAT }}
```
Closed as it does not work.

shizhMSFT reviewed (changes requested):
I've tested myself with the [Add To GitHub Projects Beta action](https://github.com/marketplace/actions/add-to-github-projects-beta).

Here are facts of it:
1. When an issue is opened, it will not have `needs-triage`. Instead, the action will add the issue to the project only if the issue is tagged `needs-triage` by someone else.
2. `GITHUB_TOKEN` does not work with error message: `Resource not accessible by integration`.
    - We should create a bot account and use the PAT of that account instead of a real person.

SteveLasker reviewed (approved):
LGTM

dtzar commented:
@shizhMSFT - I changed the ENV back to the recommended name and I also removed the tag parameter so now it will just add every new issue to the board.  I tested this configuration in my own repo with my own PAT and it worked.  So I think it's ready to be merged.  We can add in the actual PAT separately - builds for this task will just fail until it's added.

shizhMSFT reviewed (approved):
LGTM. We just need a PAT of a dedicate bot.
